### PR TITLE
Fix build warning (part 1): Add XML comments for Csharp classes

### PR DESCRIPTION
**Pull request by hebinhuang**
Add XML comments to fix the build warnings in the 5 files below:
     
ConfigurationService.cs
Accumulator.cs
Broadcast.cs
DoubleRDDFunctions.cs
Option.cs
OrderedRDDFunctions.cs

### CODE REVIEW

**Comment by danielli90** [comment]
LGTM

File: csharp/Adapter/Microsoft.Spark.CSharp/Core/Option.cs [hunk 1]
```diff
@@ -16,17 +16,32 @@ public class Option<T>
...
+        /// </summary>
+        /// <param name="value">The value to be associated with the new instance.</param>
         public Option(T value)
         {
             isDefined = true;
             this.value = value;
         }
 
+        /// <summary>
+        ///
```
Comment (skaarthik) [inline csharp/Adapter/Microsoft.Spark.CSharp/Core/Option.cs]: Missing comment?

Comment (hebinhuang) [inline csharp/Adapter/Microsoft.Spark.CSharp/Core/Option.cs]: Yes, missed it. I have added it.

**Comment by hebinhuang** [comment]
I have added the comments for Option.isDefined. Please review it again

**Comment by xiongrenyi** [comment]
a typo in Accumulator.cs ? whether -> where?
/// <summary>  
 47   +        /// Indicates whether the accumulator.  
 48   +        /// When deserialized in C# worker, isDriver is false.  
 49   +        /// </summary>  
 50   +        [NonSerialized]  
43 51            protected bool isDriver = false;

**Comment by xiongrenyi** [comment]
LGTM

**Pull request merged**